### PR TITLE
applications: asset_tracker_v2: Use new Zephyr overlay arguments

### PR DESCRIPTION
**Pull request by gregersrygg**
Replace overlay arguments OVERLAY_CONFIG and DTC_OVERLAY_FILE with the new Zephyr overlay arguments EXTRA_CONF_FILE and EXTRA_DTC_OVERLAY_FILE in documentation and sample.yaml.

This fixes a problem with some overlays overriding the board overlay for the nRF9160 DK 0.14.0 board in the application folder.

### CODE REVIEW

**Review by simensrostad — APPROVED** [review]

File: scripts/quarantine.yaml [hunk 1]
```diff
@@ -7,13 +7,6 @@
 #  platforms:
 #    - all
 
-- scenarios:
```
Comment (jtguggedal) [inline scripts/quarantine.yaml:10]: Was this intentional? 
If so, I think it should be a separate commit

Comment (gregersrygg) [reply scripts/quarantine.yaml:10]: Taking the tests out of quarantine is intentional. This PR fixes the last issue that caused these tests to fail. I can move it to a separate commit.

**Review by divipillai — APPROVED** [review]
